### PR TITLE
Modernize project tools

### PR DESCRIPTION
**Pull request by geigerzaehler**
Modernize the tools and dependencies this project uses

* Require Python ^3.8 and beets >= 1.6
* Use Poetry
* Use black and isort
* Use Github Actions

### CODE REVIEW

**Comment by wisp3rwind** [comment]
This is great!

Two small points:
- I think it would be good to keep using a test matrix with a few Python versions (oldest supported, latest stable, dev). In beets itself, this has been helpful to catch some errors due to deprecations or beets' use of Python implementation details (probably not so relevant for `beets-alternatives`, but who knows.)
- I guess the "[Add black and isort to toolchain](https://github.com/geigerzaehler/beets-alternatives/pull/70/commits/12eeaeeae1fbb4cbaf020de7cd5e78faa8e99d13)" commit has no code changes other than applying those tools?

**Comment by wisp3rwind** [comment]
Oh, and let's keep #55 open as a reminder to actually remove all Python 2 compatibility code (in particular, the `six` dependency).

**Comment by geigerzaehler** [comment]
> * I guess the "[Add black and isort to toolchain](https://github.com/geigerzaehler/beets-alternatives/pull/70/commits/12eeaeeae1fbb4cbaf020de7cd5e78faa8e99d13)" commit has no code changes other than applying those tools?

Yes, that’s correct.



> I think it would be good to keep using a test matrix with a few Python versions

Done

> Oh, and let's keep #55 open as a reminder to actually remove all Python 2 compatibility code (in particular, the `six` dependency).

I’ll close the issue and open a separate one to remove Python 2 compatibility code.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    - run: pip install poetry
+    - uses: actions/setup-python@v3
+      with:
+        python-version: ${{ matrix.python-version }}
+        cache: poetry
+    - run: poetry install
+    - run: poetry run black --check .
+    - run: poetry run isort --check .
+    - run: poetry run flake8
+    - run: poetry run pytest
```
Comment (wisp3rwind) [inline .github/workflows/main.yaml:27]: Looking at the CI logs, it seems that poetry is running via the system Python (3.10.6 at this time), and then also installs and launches beets with that python version. Do I read the logs correctly?

If so, probably the order should be changed here to run `setup-python` first, such that `pip install poetry` uses the intended Python?

Comment (geigerzaehler) [reply .github/workflows/main.yaml]: > Looking at the CI logs, it seems that poetry is running via the system Python (3.10.6 at this time), and then also installs and launches beets with that python version. Do I read the logs correctly?

You’re right. This behavior is unexpected but your probably correct that it uses the python version it was installed with. To fix this, I added the line `poetry env use $(which python)`.

> If so, probably the order should be changed here to run `setup-python` first, such that `pip install poetry` uses the intended Python?

Unfortunately, this doesn’t work. Poetry needs to be installed before `setup-python` so we can use `cache: poetry`. (See [this run](https://github.com/geigerzaehler/beets-alternatives/actions/runs/4858676611/jobs/8660367144#step:3:8) for a failure.) We could drop `cache: poetry` and write our own caching instructions but this requires more effort and may easily break.

**Review by wisp3rwind — APPROVED** [review]
:tada:

**Pull request merged**